### PR TITLE
Changelog entry for v0.6.1

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR adds a changelog entry for v0.6.1.

**Preview**: https://pygmt-pyr0d849n-gmt.vercel.app/changes.html

Previous release at #1808

Changelog made by following the instructions at https://github.com/GenericMappingTools/pygmt/blob/main/doc/maintenance.md#updating-the-changelog

TODO:

- [x] Copy the draft release note from https://github.com/GenericMappingTools/pygmt/releases to `doc/changes.md`
- [x] Update the DOI badge
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt and `CITATION.cff`
- [x] Add the documentation link https://github.com/GenericMappingTools/pygmt#documentation-for-other-versions
- [x] Add highlights
- [x] Update with any remaining PRs merged before release
- [x] Edit the change list to remove any trivial changes (updates to the README, typo fixes, CI configuration, etc)
- [x] Edit the list of people who contributed to the release, linking to their GitHub account. Sort their names by the number of commits made since the last release (e.g., use `git shortlog HEAD...v0.1.2 -sne`).

Relates to #1856

### CODE REVIEW

File: doc/changes.md [hunk 1]
```diff
@@ -1,5 +1,46 @@
 # Changelog
 
+## Release v0.6.1 (20YY/MM/DD)
+
+[![Digital Object Identifier for PyGMT v0.6.1](https://zenodo.org/badge/DOI/10.5281/zenodo.3781524.svg)](https://doi.org/10.5281/zenodo.3781524)
```
Comment (weiji14) [inline doc/changes.md:5]:
```suggestion
[![Digital Object Identifier for PyGMT v0.6.1](https://zenodo.org/badge/DOI/10.5281/zenodo.6426493.svg)](https://doi.org/10.5281/zenodo.6426493)
```

File: README.rst [hunk 2]
```diff
@@ -169,12 +169,12 @@ Developers". Feel free to cite our work in your research using the following Bib
                       Quinn, Jamie and
                       Wessel, Paul},
       title        = {{PyGMT: A Python interface for the Generic Mapping Tools}},
-      month        = march,
+      month        = april,
```
Comment (seisman) [inline README.rst:172]: I'm not sure if `april` works, but based on this page (https://www.bibtex.com/f/month-field/), we should use either `apr` or `April`
```suggestion
      month        = apr,
```

Comment (weiji14) [reply README.rst:172]: Should use `apr` I think, at least my Zotero bibliography manager uses the abbreviated form.

File: CITATION.cff [hunk 3]
```diff
@@ -64,9 +64,9 @@ authors:
   family-names: Wessel
   affiliation: University of Hawai'i at Mānoa, USA
   orcid: https://orcid.org/0000-0001-5708-7336
-date-released: 2022-03-14
-doi: 10.5281/zenodo.6349217
+date-released: 2022-04-09
```
Comment (seisman) [inline CITATION.cff:67]:
```suggestion
date-released: 2022-04-10
```

File: doc/changes.md [hunk 4]
```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+## Release v0.6.1 (2022/04/09)
```
Comment (seisman) [inline doc/changes.md:3]:
```suggestion
## Release v0.6.1 (2022/04/10)
```

**Review by michaelgrund — APPROVED** [review]
Looks good to me!

**Review by weiji14 — APPROVED** [review]
Looks good, thanks @seisman!

**Pull request merged**